dmutils/email.py

```diff
@@ -231,3 +232,95 @@ def decode_invitation_token(encoded_token):
 
 def token_created_before_password_last_changed(token_timestamp, user_timestamp):
     return token_timestamp < user_timestamp
+
+
+class DMNotifyClient(object):
+    """Digital Marketplace wrapper around the Notify python client."""
+
+    client_class = NotificationsAPIClient
```

Comment:
In my experience this paradigm makes reuse easier. What the `DMNotifyClient` class actually does is is instantiate a client and pass it arguments to send an email. 
I was going to just inherit the `NotificationsAPIClient` class but that'd be a bit fragile if there were unexpected updates.

It would be possible to do something like this:
```
class EmailerBase(object):

    client_class = None
    client_args = None


    def __init__(self):
        self.client = client_class(*client_args)
        self.logger = self.get_logger()

    def get_logger(self):
        raise NotImplementedError('Class requires get_logger')

    @staticmethod
    def decode_reference(ref, datetime_format=None):...

    @staticmethod
    def get_error_message(email_address, error):...

    def get_send_method(self):...

    def get_send_args(self):...

    def send_email(self, email_address):

        reference = self.get_reference(email_address, *self.get_send_args)
        try:
            response = self.get_send_method(
                self.get_send_args
            )
        except HTTPError as e:
            self.logger.error(self.get_error_message(email_address, e))
            raise e
        self.logger.info("Sent {email_address} template email {template_id} with Notify".format(
            email_address=email_address,
            template_id=template_id
        ))
        return response


class DMNotifyClient(EmailerBase):
    client_class = NotificationsAPIClient
    client_args = (
        govuk_notify_api_key,
        'https://api.notifications.service.gov.uk'
    )

    def get_send_method(self):
        return self.client.send_email_notification
    
    def get_send_args(self):
        return (template_id, personalisation=None, reference=None)


class MandrillClient(EmailerBase):
    client_class = Mandrill
    client_args = api_key

    def get_send_method(self):
        return self.client.messages.send

    def get_send_args(self): ...
        # get all the mandrill stuff

```